### ISSUE TITLE
Interface injection does not preserve generics

### ISSUE THREAD

**Issue by LambdAurora** [FabricMC/fabric-loom#597]
Let's say we have an event for registry additions, and a methods to get that event for a specific registry.
Now let's say this method is in an interface and we make it implement on Registry.

Registries are typed, and we have a context object, which can return the entry value, so a generic. The event would look like this:
```java
@FunctionalInterface
public interface EntryAdded<V> {
	void onAdded(RegistryEntryContext<V> context);
}
```

And the interface to implement on the registry:
```java
@FunctionalInterface
	public interface EntryAdded<V> {
		void onAdded(RegistryEntryContext<V> context);
	}
}
```

And we have a mixin for implementation:
```java
@Mixin(Registry.class)
public class RegistryMixin<V> implements RegistryEventStorage<V> {
	@Override
	public Event<RegistryEvents.EntryAdded<V>> getEntryAddedEvent() {
		return new Event<>(new Identifier("test"));
	}
}
```

So even on the mixin we have a generic type, though mixin does not seem to pay much attention to it.
Now let's reload with loom, and look at the generated Registry class file:

```java
public abstract class Registry<T> implements Keyable, IndexedIterable<T>, RegistryEventStorage {
```

This definitely does not look right, currently interface injection does not preserve any generic types.
This can be quite a problem for such cases.

Here's the stuff to replicate this: https://github.com/LambdAurora/loom_interface_injection_issue/tree/type_erasure

**Comment by modmuss50** [comment]
This is a known limitation, something that I would like to fix but not a priority right now.

**Comment by FirstMegaGame4** [comment]
This issue should be marked as completed as it has been solved in #1050.
